### PR TITLE
eNikshay custom reports: Display location restricted users' locations in location filter

### PR DESCRIPTION
**Pull request by NoahCarnahan**
When a user loads a custom eNikshay report with a location filter, if that user is location restricted, their location will be displayed as a selected value in that report. (The report is already filtered by the users location, but previously there had been no visible indication of this).
@esoergel 
buddy @gcapalbo

### CODE REVIEW

File: custom/enikshay/reports/filters.py [hunk 1]
```diff
@@ -32,6 +32,11 @@ def selected(self):
         :return: Selected locations without their descendants
         """
         location_ids = self.request.GET.getlist(self.slug)
+
+        if len(location_ids) == 0 and not self.request.couch_user.has_permission(self.request.domain, 'access_all_locations'):
```
Comment (dimagimon) [inline custom/enikshay/reports/filters.py]: E501 line too long (126 > 115 characters)

Comment (dimagimon) [inline custom/enikshay/reports/filters.py]: E501 line too long (126 > 115 characters)

Comment (NoahCarnahan) [inline custom/enikshay/reports/filters.py]: dimagimon is confused, this is not what the diff looks like any more

File: custom/enikshay/reports/filters.py [hunk 2]
```diff
@@ -32,6 +32,12 @@ def selected(self):
         :return: Selected locations without their descendants
         """
         location_ids = self.request.GET.getlist(self.slug)
+
+        if len(location_ids) == 0 \
+                and not self.request.couch_user.has_permission(self.request.domain, 'access_all_locations'):
+            # Display the user's location in the filter if none is selected
+            location_ids = self.request.couch_user.get_location_ids(self.request.domain)
+
```
Comment (esoergel) [inline custom/enikshay/reports/filters.py]: Nice, I like this idea.  `request.can_access_all_locations` is also available as long as it's a real request (not like, a test mock)

**Pull request merged**